syserol/model.py

```diff
@@ -106,40 +107,42 @@ def test_predictions(function="ADCD"):
     return corr
 
 
+@lru_cache()
 def cross_validation():
     """ 10 Fold Cross Validation to Test Predictive Abilities"""
     cube, glyCube = createCube()
-    _, mapped = importFunction()
+    _, mapped = importFunction() 
     glycan, _ = importGlycan()
 
     X = glyCube
+    copy = glyCube #copy glyCube
     index = []
     original = []
     predicted = []
     kf = KFold(n_splits=10, shuffle=True) # split into 10 folds
     for train_index, test_index in kf.split(X): # run cross validation
-        for i in test_index:
+        glyCube = copy # restore original values at start of each cross validation fold
+        for i in test_index: # for all the selected test indices 
             for j, _ in enumerate(mapped):
-                index.append((i, len(glycan) + j))
-                original.append(glyCube[i][len(glycan) + j])
-                glyCube[i][len(glycan) + j] = np.nan
```

Comment:
Also, you have to assemble the predicted values within the loop, because you're leaving out a few in each fold, but the originals value is just a slice of the tensor. You shouldn't need to assemble that within the loop.

Reply:
I have this running in the loop because I set shuffle=True, meaning the test indices are chosen randomly within each loop, and I want to make sure I'm selecting the correct section of the tensor that I had left out.